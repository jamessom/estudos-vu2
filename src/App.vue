<template>
  <div id="app">
    <img alt="Vue logo" src="./assets/logo.png">
    <HelloWorld msg="Welcome to Your Vue.js App"/>
    <conditional-and-lists></conditional-and-lists>
  </div>
</template>

<script>
import HelloWorld from '@/components/HelloWorld.vue'
import ConditionalAndLists from '@/components/ConditionalAndLists.vue'

export default {
  name: 'App',
  components: {
    HelloWorld,
    ConditionalAndLists
  }
}
</script>

<style>
#app {
  font-family: 'Avenir', Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
</style>
