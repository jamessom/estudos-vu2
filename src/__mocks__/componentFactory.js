import { shallowMount, createLocalVue } from '@vue/test-utils';
import VueRouter from '@/router/index';

export default function componentFactory(options = {}) {
  return {
    wrapper: () => {
      const { component, componentOptions, withRouter } = options;
      const localvue = createLocalVue();

      if (withRouter) {
        localvue.use(VueRouter);

        return shallowMount(component, {
          localvue,
          ...componentOptions,
        });
      }

      return shallowMount(component, {
        localvue,
        ...componentOptions,
      });
    },
  };
}
