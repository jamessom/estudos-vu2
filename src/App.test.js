import componentFactory from '@/__mocks__/componentFactory';
import App from './App.vue';

describe('src/App.vue', () => {
  describe('when component is mounted', () => {
    const snapshotTest = componentFactory({
      component: App,
      withRouter: true,
    });

    it('should renders correctly', () => {
      expect(snapshotTest.wrapper().element).toMatchSnapshot();
    });
  });
});
