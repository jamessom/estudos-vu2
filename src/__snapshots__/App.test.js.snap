// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/App.vue when component is mounted should renders correctly 1`] = `
<div
  id="app"
>
  <router-view-stub
    name="default"
  />
</div>
`;
